dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1141,6 +1158,40 @@ CDefaultContextMenu::InvokeRegVerb(
 
     PStaticShellEntry pEntry = &m_StaticEntries.GetAt(it);
 
+    WCHAR VerbKey[sizeof("shell\\") + MAX_VERB];
+    hr = StringCbPrintfW(VerbKey, sizeof(VerbKey), L"shell\\%s", pEntry->Verb.GetString());
+    HKEY hVerbKey;
+    if (SUCCEEDED(hr) && m_pDataObj &&
+        RegOpenKeyExW(pEntry->hkClass, VerbKey, 0, KEY_READ, &hVerbKey) == ERROR_SUCCESS)
+    {
+        DWORD KeyState = 0;
+        if (lpcmi->fMask & CMIC_MASK_SHIFT_DOWN)
+            KeyState |= MK_SHIFT;
+        if (lpcmi->fMask & CMIC_MASK_CONTROL_DOWN)
+            KeyState |= MK_CONTROL;
+        POINTL *pPtl = NULL;
+        C_ASSERT(sizeof(POINT) == sizeof(POINTL));
+        if (lpcmi->fMask & CMIC_MASK_PTINVOKE)
+            pPtl = (POINTL*)&lpcmi->ptInvoke;
+        CLSID clsid;
+
+        // TODO: IExecuteCommand
```

Comment:
Would this future IExecuteCommand use the clsid variable introduced above?

Reply:
Yes.